src/core/tests/type_prop/unsqueeze.cpp

```diff
@@ -62,35 +80,193 @@ TEST(type_prop, unsqueeze_empty_axes) {
+    unsqueeze->set_arguments(NodeVector{param, axes_node});
+    unsqueeze->validate_and_infer_types();
+
+    EXPECT_EQ(unsqueeze->get_output_element_type(0), element::f32);
+    EXPECT_EQ(unsqueeze->get_output_partial_shape(0), exp_shape);
+}
+
+using BoundTestParam = std::tuple<PartialShape, PartialShape>;
+
+class UnsqueezeBoundTest : public WithParamInterface<BoundTestParam>, public UnsqueezeTestCommon {
+protected:
+    void SetUp() override {
+        std::tie(p_shape, exp_shape) = GetParam();
+        UnsqueezeTestCommon::SetUp();
+    }
 
+    std::vector<size_t> exp_labels;
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    type_prop_bounds_propagate,
+    UnsqueezeBoundTest,
+    Values(std::make_tuple(PartialShape::dynamic(2), PartialShape::dynamic(1)),
+           std::make_tuple(PartialShape{Dimension(-1)}, PartialShape{Dimension(-1)}),
+           std::make_tuple(PartialShape{Dimension::dynamic(), 8}, PartialShape{Dimension::dynamic()}),
+           std::make_tuple(PartialShape{Dimension(2, 5), Dimension::dynamic()}, PartialShape{Dimension(2, 5)}),
+           std::make_tuple(PartialShape{Dimension(2, -1), Dimension::dynamic()}, PartialShape::dynamic(1)),
+           std::make_tuple(PartialShape{Dimension(-1, 3), Dimension::dynamic()}, PartialShape{Dimension(-1, 3)}),
+           std::make_tuple(PartialShape{5}, PartialShape{5}),
+           std::make_tuple(PartialShape{2, 6}, PartialShape{2})),
+    PrintToStringParamName());
+
+/**
+ * \brief Check label and dynamic value propagation.
+ *
+ * Test use evaluate label, lower/upper.
+ */
+TEST_P(UnsqueezeBoundTest, propagate_label_and_dynamic_value) {
+    PartialShape labeled_shape = PartialShape{p_shape};
+
+    std::generate_n(std::back_inserter(exp_labels), labeled_shape.size(), ov::SeqGen<size_t>(1));
+    set_shape_labels(labeled_shape, exp_labels);
+
+    constexpr auto et = element::i64;
+    const auto labeled_param = std::make_shared<op::Parameter>(et, labeled_shape);
+    const auto labeled_shape_of = std::make_shared<op::ShapeOf>(labeled_param);
+
+    const auto zero = std::vector<int64_t>{0};
+    const auto axis = std::make_shared<op::v0::Constant>(et, Shape{}, zero);
+    const auto indices = std::make_shared<op::v0::Constant>(et, Shape{}, zero);
+    const auto gather = std::make_shared<op::v7::Gather>(labeled_shape_of, indices, axis);
     const auto unsqueeze = std::make_shared<op::v0::Unsqueeze>(gather, axis);
 
-    auto bc = std::make_shared<op::v1::Broadcast>(param, unsqueeze);
-    ASSERT_EQ(bc->get_shape(), (Shape{3}));
+    const auto bc = std::make_shared<op::v1::Broadcast>(param, unsqueeze);
 
-    const auto& output_shape = bc->get_output_partial_shape(0);
-    ASSERT_EQ(ov::DimensionTracker::get_label(output_shape[0]), 10);
+    EXPECT_EQ(bc->get_output_partial_shape(0), exp_shape);
+    const auto labels = get_shape_labels(bc->get_output_partial_shape(0));
+    EXPECT_THAT(labels, ElementsAre(exp_labels.front()));
```

Comment:
It's a little bit hard to follow the logic and shapes used in this test, and what the expected labels really are in this case.
Additional test case with single `Unsqueeze` and example of input shape and expected labels could be added.

Reply:
It expect that labels got one element which got value of first from input (expected) labels, no sure what is not clear.

Added tests for label propagation for single `Unsqueeze`